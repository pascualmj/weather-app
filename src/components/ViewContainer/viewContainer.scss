.view-container {
  width: 100%;
  max-width: 66rem;
  margin: 15% auto;
  padding: 0 3rem;
}
