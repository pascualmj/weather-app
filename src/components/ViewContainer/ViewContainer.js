import React from "react";
import PropTypes from "prop-types";
import "./viewContainer.scss";

const ViewContainer = ({ children }) => {
  return <main className="view-container">{children}</main>;
};

ViewContainer.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ViewContainer;
