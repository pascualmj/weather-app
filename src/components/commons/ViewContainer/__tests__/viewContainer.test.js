import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ViewContainer from "../ViewContainer";

describe("Component: ViewContainer", () => {
  test("Should match snapshot.", () => {
    const { asFragment } = render(<ViewContainer>lorem</ViewContainer>);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render children.", () => {
    const { getByText } = render(<ViewContainer>hi!</ViewContainer>);

    expect(getByText(/hi!/i)).not.toBeNull();
  });
});
