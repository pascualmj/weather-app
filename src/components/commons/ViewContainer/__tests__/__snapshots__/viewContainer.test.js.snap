// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ViewContainer Should match snapshot. 1`] = `
<DocumentFragment>
  <main
    class="view-container"
  >
    lorem
  </main>
</DocumentFragment>
`;
